src/util.lua

```diff
@@ -283,4 +334,19 @@ function util.extract_keys(nspace, tokens)
   return keys
 end
 
+-- store argument list for or return values of a function in a table
+function util.make_arglist(...)
+  local arglist = { ... }
+  arglist.n = select('#', ...) -- add values count for trailing nils
+  return setmetatable(arglist, arglist_mt)
+end
+
+function util.is_arglist(object)
+  return type(object) == "table" and getmetatable(object) == arglist_mt
```

Comment:
the first check (for being a table) is not necessary, every type has a metatable

Reply:
I probably copied this from a similar function in spy.lua or match.lua but will remove it in an updated commit.